transportenergy/database
# Update CI and tests

Various housekeeping changes:

- Adjust code to satisfy mypy with latest version and dependencies.
- Update "pytest" CI workflow:
  - Split the "diagnostics" to a separate job.
  - Add a code quality/pre-commit job.
  - Update third-party actions.
  - Run on all 3 supported operating systems and Python up to 3.12.
- Convert packaging from setup.{cfg,py} to pyproject.toml.
- Use ruff and mypy via pre-commit for code quality checks.
- Adjust for "Turkey" no longer appearing in the ISO 3166 database.
- Xfail certain tests: in particular, those using data via OpenKAPSARC that are no longer available because the provider has removed them.
